app/src/main/java/dev/sebastiano/bundel/storage/DatabaseModule.kt

```diff
@@ -2,22 +2,27 @@ package dev.sebastiano.bundel.storage
 
 import android.app.Application
 import androidx.room.Room
+import dagger.Binds
 import dagger.Module
 import dagger.Provides
 import dagger.hilt.InstallIn
 import dagger.hilt.components.SingletonComponent
 import dev.sebastiano.bundel.storage.migrations.Migration1to2
+import javax.inject.Singleton
 
 @Module
 @InstallIn(SingletonComponent::class)
-internal class DatabaseModule {
+internal abstract class DatabaseModule {
 
-    @Provides
-    fun provideImagesStorage(application: Application): ImagesStorage = DiskImagesStorage(application)
+    @Binds
+    abstract fun provideImagesStorage(diskImagesStorage: DiskImagesStorage): ImagesStorage
 
-    @Provides
-    fun provideDatabase(application: Application): RobertoDatabase =
-        Room.databaseBuilder(application, RobertoDatabase::class.java, "roberto")
-            .addMigrations(Migration1to2)
-            .build()
+    companion object {
```

Comment:
Is there a specific reason not to make `DatabaseModule` an `object` instead? Or do we need the `abstract class` for Hilt/Dagger to work?

Reply:
The `DatabaseModule` is abstract because it has an abstract method, and `@Binds` methods must be abstract. Using `@Binds` and `@Provides` methods in the same module is mentioned [here](https://github.com/google/dagger/releases/tag/dagger-2.26). Now, whether it's possible to have an object module, I don't know.

Reply:
Got it, thanks for the explanation :)